### PR TITLE
LPS-153204 Display what type of basic document is being shown in the Info Panel of the Content Dashboard

### PR DESCRIPTION
**Pull request by cgoncas**
### Motivation

As a content marketer, I want to see the type of document in the info panel, so that I can better understand the asset I am previewing.

More info in the story: [LPS-153204](https://issues.liferay.com/browse/LPS-153204)

### Proposed Solution

1. Create a new API ContentDashboardItemFilter. This api will define a new filter to be added to the Filter in Content Dashboard (Only Item Selectors will be supported): https://github.com/liferay-tango/liferay-portal/pull/2674
2. Implement the ContentDashboardItemFilter API in the documents and media module, so we can uncouple the code from content-dashboard-web module: https://github.com/liferay-tango/liferay-portal/pull/2683
3. Refactor the code in Content-Dashboard-Web so we can uncouple the code completely.
5. Move the Subtype to the API. https://github.com/liferay-tango/liferay-portal/pull/2685
4. Use the FileExtensionGroup code to show the right file type.  https://github.com/liferay-tango/liferay-portal/pull/2727

### CODE REVIEW

**Comment by cgoncas** [comment]
Hey @beltranrengifo and @claraizquierdo !!

I am trying to move the fileExtension code from the content-dashboard-web to content-dashboard-document-library-impl. Now the JSP is being loaded succesfully, but the JS/CSS is not working, could you help me with that part?

Thanks!!

**Comment by beltranrengifo** [comment]
> Hey @beltranrengifo and @claraizquierdo !!
> 
> I am trying to move the fileExtension code from the content-dashboard-web to content-dashboard-document-library-impl. Now the JSP is being loaded succesfully, but the JS/CSS is not working, could you help me with that part?
> 
> Thanks!!

Sure! Can you give me a bit more of context so I can get to it easier?

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by beltranrengifo** [comment]
hi @cgoncas !
the CD web module is not compiling, I imagine you are aware, just want to confirm we see the same:

```
> Task :apps:content-dashboard:content-dashboard-web:compileJSP
/Users/beltranrengifo/Dev/Liferay/lr-portal/liferay-portal/modules/apps/content-dashboard/content-dashboard-web/build/jspc/org/apache/jsp/view_005fcontent_005fdashboard_005ffile_005fextensions_jsp.java:194: error: cannot find symbol
ContentDashboardFileExtensionItemSelectorViewDisplayContext contentDashboardFileExtensionItemSelectorViewDisplayContext = (ContentDashboardFileExtensionItemSelectorViewDisplayContext)request.getAttribute(ContentDashboardFileExtensionItemSelectorViewDisplayContext.class.getName());
```

**Comment by cgoncas** [comment]
Hey @beltranrengifo !!

I have found that a view was not being registered I don't know why, I am working on it :). we are closer.

Thanks!!

**Comment by cgoncas** [comment]
> Hey @beltranrengifo !!
> 
> I have found that a view was not being registered I don't know why, I am working on it :). we are closer.
> 
> Thanks!!

Hey @beltranrengifo !!

I am receiving the error:

`2022-08-08 09:26:04.887 WARN  [http-nio-8080-exec-9][code_jsp:163] {code="404", msg="Not Found", uri=/o/content-dashboard-document-library-impl/css/tree.css}
`

Do you know what should we add in order to load the CSS propertly?

Thanks!!

**Comment by cgoncas** [comment]
> hi @cgoncas ! the CD web module is not compiling, I imagine you are aware, just want to confirm we see the same:
> 
> ```
> > Task :apps:content-dashboard:content-dashboard-web:compileJSP
> /Users/beltranrengifo/Dev/Liferay/lr-portal/liferay-portal/modules/apps/content-dashboard/content-dashboard-web/build/jspc/org/apache/jsp/view_005fcontent_005fdashboard_005ffile_005fextensions_jsp.java:194: error: cannot find symbol
> ContentDashboardFileExtensionItemSelectorViewDisplayContext contentDashboardFileExtensionItemSelectorViewDisplayContext = (ContentDashboardFileExtensionItemSelectorViewDisplayContext)request.getAttribute(ContentDashboardFileExtensionItemSelectorViewDisplayContext.class.getName());
> ```

About that error, you just need to run `git clean -fdx  ` on content-dashboard-web, something has been cached there :)

**Comment by beltranrengifo** [comment]
> Hey @beltranrengifo !!
> 
> I am receiving the error:
> 
> `2022-08-08 09:26:04.887 WARN [http-nio-8080-exec-9][code_jsp:163] {code="404", msg="Not Found", uri=/o/content-dashboard-document-library-impl/css/tree.css} `
> 
> Do you know what should we add in order to load the CSS propertly?
> 
> Thanks!!

Hmmm i see we are duplicating the js components and utils... not good
Is there a way we could load the js part from the web module?

**Comment by beltranrengifo** [comment]
We're gonna try to reuse the js/css from the web module exporting a new package.
We'll let you know asap

**Comment by beltranrengifo** [comment]
the main error we saw before gets solved just by creating the package.json for the module, this way it gets compiled by the bundler, os something like that 🤣

**Comment by cgoncas** [comment]
> the main error we saw before gets solved just by creating the package.json for the module, this way it gets compiled by the bundler, something like that I guess 🤣

Hey @beltranrengifo !!

So now the JS is being loaded propertly?

Thanks!!

**Comment by beltranrengifo** [comment]
> 

Yes it is.
But Im trying to reuse the JS from the web module. I also have to figure out what to do with the CSS

**Comment by cgoncas** [comment]
> > 
> 
> Yes it is. But Im trying to reuse the JS from the web module. I also have to figure out what to do with the CSS

That sounds awesome!!!

**Comment by beltranrengifo** [comment]
I'm pushing some changes. It's working fine here. Let me know if it works for you...
cc @claraizquierdo

File: modules/apps/content-dashboard/content-dashboard-document-library-impl/src/main/resources/META-INF/resources/view_content_dashboard_file_extensions.jsp [hunk 1]
```diff
@@ -20,12 +20,9 @@
 ContentDashboardFileExtensionItemSelectorViewDisplayContext contentDashboardFileExtensionItemSelectorViewDisplayContext = (ContentDashboardFileExtensionItemSelectorViewDisplayContext)request.getAttribute(ContentDashboardFileExtensionItemSelectorViewDisplayContext.class.getName());
 %>
 
-<liferay-util:html-top>
-	<link href="<%= PortalUtil.getStaticResourceURL(request, application.getContextPath() + "/css/tree.css") %>" rel="stylesheet" type="text/css" />
-</liferay-util:html-top>
-
 <section class="h-100">
-	<span aria-hidden="true" class="loading-animation mt-0 tree-filter-loader"></span>
+	<span aria-hidden="true" class="loading-animation mt-0 tree-filter-loader"
```
Comment (beltranrengifo) [inline modules/apps/content-dashboard/content-dashboard-document-library-impl/src/main/resources/META-INF/resources/view_content_dashboard_file_extensions.jsp:24]: I think we can delete that css class `tree-filter-loader` as it wont get load at this point.

I'm moving just this styles inline so we do not have to mess around with the whole CSS file.
The rest of the styles are loaded fine by the JS

**Comment by beltranrengifo** [comment]
Whoops, the title is wrong, I'll take a look...

image

**Comment by cgoncas** [comment]
> Whoops, the title is wrong, I'll take a look...
> 
> image

I think this is my fault, in the configuration of the item selector.

**Comment by claraizquierdo** [comment]
> I'm pushing some changes. It's working fine here. Let me know if it works for you... cc @claraizquierdo

Ei @beltranrengifo , I like having all the React components under content-dashboard-web, but I see one "problem" with this solution: we are defining SelectFileExtension component in the content-dashboard-web but we are not using it there.

**Comment by beltranrengifo** [comment]
> > Whoops, the title is wrong, I'll take a look...
> > image
> 
> I think this is my fault, in the configuration of the item selector.

Could it be provoked by an incorrect lang key?

**Comment by cgoncas** [comment]
> > > Whoops, the title is wrong, I'll take a look...
> > > image
> > 
> > 
> > I think this is my fault, in the configuration of the item selector.
> 
> Could it be provoked by an incorrect lang key?

I think so.

**Comment by beltranrengifo** [comment]
> SelectFileExtension

I push it!

**Comment by beltranrengifo** [comment]
> > I'm pushing some changes. It's working fine here. Let me know if it works for you... cc @claraizquierdo
> 
> Ei @beltranrengifo , I like having all the React components under content-dashboard-web, but I see one "problem" with this solution: we are defining SelectFileExtension component in the content-dashboard-web but we are not using it there.

You're right @claraizquierdo 
This component was referenced only in the JSP file, and now that file is in `content-dashboard-document-library-impl`
Would be ideal to define it closer to the usage, but as long as it depends on other files from web module, I think is the best or _less bad_ solution... What are you thinking about?

**Comment by beltranrengifo** [comment]
> > I'm pushing some changes. It's working fine here. Let me know if it works for you... cc @claraizquierdo
> 
> Ei @beltranrengifo , I like having all the React components under content-dashboard-web, but I see one "problem" with this solution: we are defining SelectFileExtension component in the content-dashboard-web but we are not using it there.

Being strict, it's used [here](https://github.com/liferay-tango/liferay-portal/blob/8f41eb308a8d9b5eee2375f127802437f6e2291b/modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/index.js) 😀

**Comment by claraizquierdo** [comment]
> > > I'm pushing some changes. It's working fine here. Let me know if it works for you... cc @claraizquierdo
> > 
> > 
> > Ei @beltranrengifo , I like having all the React components under content-dashboard-web, but I see one "problem" with this solution: we are defining SelectFileExtension component in the content-dashboard-web but we are not using it there.
> 
> Being strict, it's used [here](https://github.com/liferay-tango/liferay-portal/blob/8f41eb308a8d9b5eee2375f127802437f6e2291b/modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/index.js) 😀

If we can't keep the view_content_dashboard_file_extensions.jsp file inside content-dashboard-web I think the best solution is what you did Beltran 👍

**Comment by cgoncas** [comment]
Closing this PR, because all the code has been sent to Brian Chan

**Pull request closed without merging**